### ISSUE TITLE
Regression in bundler 2.5.3

### ISSUE THREAD

**Issue by jcoyne** [rubygems/rubygems#7357]
Bundler no longer bundles development dependencies listed in gemspec

I'm seeing:
```
Failure/Error: require 'simplecov'

LoadError:
  cannot load such file -- simplecov
# ./spec/spec_helper.rb:9:in `require'
# ./spec/spec_helper.rb:9:in `<top (required)>'
No examples found.
```

You can see that in this PR I have one commit using 2.5.2 where it works as expected and a commit where it fails using 2.5.3:
https://github.com/projectblacklight/blacklight/pull/3117

I think this means #7319 is the culprit since it is the only change in 2.5.3 according to https://github.com/rubygems/rubygems/commit/9f9ba2108d3f83f5aad3b83a2341f481ff2c3495

Curiously I do see the message:
```
A gemspec development dependency (rsolr, >= 1.0.6, < 3) is being overridden by a Gemfile dependency (rsolr, >= 1.0, < 3).
```

But there is nothing about `simplecov`.

**Comment by deivid-rodriguez** [comment]
Funny I had just discovered an issue that seems similar. Can you check if the change in #7358 fixes it?

**Comment by jcoyne** [comment]
@deivid-rodriguez I'm not quite certain how you can install bundler/rubygems from source in order to test it out.

**Comment by deivid-rodriguez** [comment]
You'd need to clone the repo, run `rake install:bundler`, and make sure your lockfile is locked to `2.6.0.dev`.

**Comment by jcoyne** [comment]
@deivid-rodriguez yes, using that branch fixes the problem. Thanks for the help with the instructions.

**Comment by deivid-rodriguez** [comment]
Fixed by #7358.
